sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/client.py

```diff
@@ -523,109 +549,36 @@ def _send_message_impl(self, message, **kwargs):
-        - 2. Transform the array to AMQPValue object: `utils.data_factory(types.AMQPArray(capabilities_symbol_array))`
-    :type desired_capabilities: ~uamqp.c_uamqp.AMQPValue
-    :param max_message_size: The maximum allowed message size negotiated for the Link.
-    :type max_message_size: int
-    :param link_properties: Metadata to be sent in the Link ATTACH frame.
-    :type link_properties: dict
-    :param prefetch: The receiver Link credit that determines how many
-     messages the Link will attempt to handle per connection iteration.
-     The default is 300.
-    :type prefetch: int
-    :param max_frame_size: Maximum AMQP frame size. Default is 63488 bytes.
-    :type max_frame_size: int
-    :param channel_max: Maximum number of Session channels in the Connection.
-    :type channel_max: int
-    :param idle_timeout: Timeout in seconds after which the Connection will close
-     if there is no further activity.
-    :type idle_timeout: int
-    :param properties: Connection properties.
-    :type properties: dict
-    :param remote_idle_timeout_empty_frame_send_ratio: Ratio of empty frames to
-     idle time for Connections with no activity. Value must be between
-     0.0 and 1.0 inclusive. Default is 0.5.
-    :type remote_idle_timeout_empty_frame_send_ratio: float
-    :param incoming_window: The size of the allowed window for incoming messages.
-    :type incoming_window: int
-    :param outgoing_window: The size of the allowed window for outgoing messages.
-    :type outgoing_window: int
-    :param handle_max: The maximum number of concurrent link handles.
-    :type handle_max: int
-    :param on_attach: A callback function to be run on receipt of an ATTACH frame.
-     The function must take 4 arguments: source, target, properties and error.
-    :type on_attach: func[~uamqp.address.Source, ~uamqp.address.Target, dict, ~uamqp.errors.AMQPConnectionError]
-    :param encoding: The encoding to use for parameters supplied as strings.
-     Default is 'UTF-8'
-    :type encoding: str
-    """
 
-    def __init__(self, hostname, source, auth=None, **kwargs):
+    def __init__(self, hostname, source, **kwargs):
         self.source = source
-        self._streaming_receive = kwargs.pop("streaming_receive", False)  # TODO: whether public?
+        self._streaming_receive = kwargs.pop("streaming_receive", False) 
         self._received_messages = queue.Queue()
-        self._message_received_callback = kwargs.pop("message_received_callback", None)  # TODO: whether public?
+        self._message_received_callback = kwargs.pop("message_received_callback", None) 
 
         # Sender and Link settings
-        self._max_message_size = kwargs.pop('max_message_size', None) or MAX_FRAME_SIZE_BYTES
+        self._max_message_size = kwargs.pop('max_message_size', MAX_FRAME_SIZE_BYTES)
         self._link_properties = kwargs.pop('link_properties', None)
         self._link_credit = kwargs.pop('link_credit', 300)
```

Comment:
There is a default_link_credit const, perhaps add in default_receive_link_credit?